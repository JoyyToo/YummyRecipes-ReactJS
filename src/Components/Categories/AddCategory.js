import React, { Component }  from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import TextField  from 'material-ui/TextField';
import { Link } from 'react-router-dom';
import '../styles.css'
import Paper from 'material-ui/Paper';
import {notify} from 'react-notify-toast'
import axiosInstance from '../Constants/AxiosCall';


class AddCategory extends Component {
    // initialize state
    state = {
        name: '',
        desc: ''
    }

    // handle user input
    handleInputChange = (event) => {
        const target = event.target;
        const value = target.value;
        const name = target.name;
        this.setState({[name]: value});
    }

    // mount token when page loads
    componentDidMount() {
        const token = window.localStorage.getItem('token');
        if (!token) {
            window.location.replace('/login')
        }
    }

    // handle add category request
    handleAddcategory = (event) => {
    const payload = new FormData()
    payload.set('name', this.state.name)    
    payload.set('desc', this.state.desc,)
    const token = window.localStorage.getItem('token')
    if (!token) {
        window.location.replace('/login')
    }
        
        // send POST request to API
        axiosInstance.post('category', payload)
        .then((response) => {
            notify.show(response.data.message, 'success', 4000);
            this.props
            .history
            .push('/categories');
        })
        .catch((error) => {
            notify.show(error.response.data.message, 'error', 4000);
        });
    }

    // render add category form
    render(){
        let style = {
            marginLeft: 20,
            width: 340,
          };
        let divstyle = {
            width: 430,
            margin: 'auto',
            textAlign: 'left',
            paddingLeft: 20,
        }
        return (
            <MuiThemeProvider>
            <div className="main">
                <div className="inputs" style={{paddingTop: '150px'}}>
                <Paper zDepth={2} style={divstyle}>
                    <p class="heading"><b>ADD CATEGORY</b></p>

                    <TextField floatingLabelText="Name" name="name" 
                                 value={this.state.name} style={style} onChange={this.handleInputChange} /><br /><br />

                    <TextField floatingLabelText="Description" name="desc" 
                                 value={this.state.desc} style={style} onChange={this.handleInputChange} /><br /><br /><br />

                    <div className="buttons">
                    <button className="network" id="one" onClick={(event => this.handleAddcategory(event))}>ADD</button>
                    <Link to="/categories"><button className="network" id="two">CANCEL</button></Link><br /><br /><br />

                    </div>

                </Paper><br />
                    
                </div>
            </div>
            </MuiThemeProvider>
        );
    }}

export default AddCategory;
