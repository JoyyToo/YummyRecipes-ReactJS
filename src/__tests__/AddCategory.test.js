import React from 'react';
<<<<<<< HEAD
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
=======
import { shallow, mount } from 'enzyme';
import toJson, { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
>>>>>>> [Feature 155119223]:

import AddCategory from '../Components/Categories/Addcategory'

describe('AddCategory component', () => {
    const wrapper = shallow(<AddCategory />);
  
    it('renders properly', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders 1 <AddCategory/> component', () =>{
        const component = shallow(<AddCategory/>);
        expect(component).toHaveLength(1);
    });

    it('it returns props correctly', () =>{
        const component = shallow(<AddCategory name="app"/>);
        expect(component.instance().props.name).toBe('app')
    });

<<<<<<< HEAD
    it('contains MuiThemeProvider', () => {
        expect(wrapper.find('MuiThemeProvider')).toHaveLength(1);
    });

    it('contains divs', () => {
        expect(wrapper.find('div')).toHaveLength(3);
    });

    it('contains Paper', () => {
        expect(wrapper.find('Paper')).toHaveLength(1);
    });

    it('contains p', () => {
        expect(wrapper.find('p')).toHaveLength(1);
    });

    it('contains TextField ', () => {
        expect(wrapper.find('TextField')).toHaveLength(2);
    });

    it('contains button', () => {
        expect(wrapper.find('button')).toHaveLength(2);
    });

    it('contains Link', () => {
        expect(wrapper.find('Link')).toHaveLength(1);
    });

    it('contains br', () => {
        expect(wrapper.find('br')).toHaveLength(9);
    });

    it('contains b', () => {
=======
    it('renders component in MuiThemeProvider', () => {
        expect(wrapper.find('MuiThemeProvider')).toHaveLength(1);
    });

    it('renders component in divs', () => {
        expect(wrapper.find('div')).toHaveLength(3);
    });

    it('renders component in Paper', () => {
        expect(wrapper.find('Paper')).toHaveLength(1);
    });

    it('renders component in p', () => {
        expect(wrapper.find('p')).toHaveLength(1);
    });

    it('renders component in TextField ', () => {
        expect(wrapper.find('TextField')).toHaveLength(2);
    });

    it('renders component in button', () => {
        expect(wrapper.find('button')).toHaveLength(2);
    });

    it('renders component in Link', () => {
        expect(wrapper.find('Link')).toHaveLength(1);
    });

    it('renders component in br', () => {
        expect(wrapper.find('br')).toHaveLength(9);
    });

    it('renders component in b', () => {
>>>>>>> [Feature 155119223]:
        expect(wrapper.find('b')).toHaveLength(1);
    });

    it('has initial state', () => {
        expect(wrapper.state().name).toEqual('');
        expect(wrapper.state().desc).toEqual('');
    });

    it('changes state', () => {
        wrapper.setState({ name: 'dessert', desc: 'after meals' });
        expect(wrapper.find('[name="name"]').props().value).toEqual('dessert');
        expect(wrapper.find('[name="desc"]').props().value).toEqual('after meals');
    });

    it('has the correct form fields', () => {
        expect(wrapper.find('[name="name"]')).toHaveLength(1);
        expect(wrapper.find('[name="desc"]')).toHaveLength(1);
    });

})