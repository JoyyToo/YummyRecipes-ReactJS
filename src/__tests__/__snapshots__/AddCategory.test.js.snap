// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory component renders properly 1`] = `
<MuiThemeProvider>
  <div
    className="main"
  >
    <div
      className="inputs"
      style={
        Object {
          "paddingTop": "150px",
        }
      }
    >
      <Paper
        circle={false}
        rounded={true}
        style={
          Object {
            "margin": "auto",
            "paddingLeft": 20,
            "textAlign": "left",
            "width": 430,
          }
        }
        transitionEnabled={true}
        zDepth={2}
      >
        <p
          class="heading"
        >
          <b>
            ADD CATEGORY
          </b>
        </p>
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Name"
          fullWidth={false}
          multiLine={false}
          name="name"
          onChange={[Function]}
          rows={1}
          style={
            Object {
              "marginLeft": 20,
              "width": 340,
            }
          }
          type="text"
          underlineShow={true}
          value=""
        />
        <br />
        <br />
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Description"
          fullWidth={false}
          multiLine={false}
          name="desc"
          onChange={[Function]}
          rows={1}
          style={
            Object {
              "marginLeft": 20,
              "width": 340,
            }
          }
          type="text"
          underlineShow={true}
          value=""
        />
        <br />
        <br />
        <br />
        <div
          className="buttons"
        >
          <button
            className="network"
            id="one"
            onClick={[Function]}
          >
            ADD
          </button>
          <Link
            replace={false}
            to="/categories"
          >
            <button
              className="network"
              id="two"
            >
              CANCEL
            </button>
          </Link>
          <br />
          <br />
          <br />
        </div>
      </Paper>
      <br />
    </div>
  </div>
</MuiThemeProvider>
`;

exports[`AddCategory component renders properly 1`] = `
<MuiThemeProvider>
  <div
    className="main"
  >
    <div
      className="inputs"
      style={
        Object {
          "paddingTop": "150px",
        }
      }
    >
      <Paper
        circle={false}
        rounded={true}
        style={
          Object {
            "margin": "auto",
            "paddingLeft": 20,
            "textAlign": "left",
            "width": 430,
          }
        }
        transitionEnabled={true}
        zDepth={2}
      >
        <p
          class="heading"
        >
          <b>
            ADD CATEGORY
          </b>
        </p>
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Name"
          fullWidth={false}
          multiLine={false}
          name="name"
          onChange={[Function]}
          rows={1}
          style={
            Object {
              "marginLeft": 20,
              "width": 340,
            }
          }
          type="text"
          underlineShow={true}
          value=""
        />
        <br />
        <br />
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Description"
          fullWidth={false}
          multiLine={false}
          name="desc"
          onChange={[Function]}
          rows={1}
          style={
            Object {
              "marginLeft": 20,
              "width": 340,
            }
          }
          type="text"
          underlineShow={true}
          value=""
        />
        <br />
        <br />
        <br />
        <div
          className="buttons"
        >
          <button
            className="network"
            id="one"
            onClick={[Function]}
          >
            ADD
          </button>
          <Link
            replace={false}
            to="/categories"
          >
            <button
              className="network"
              id="two"
            >
              CANCEL
            </button>
          </Link>
          <br />
          <br />
          <br />
        </div>
      </Paper>
      <br />
    </div>
  </div>
</MuiThemeProvider>
`;
